colearn_keras/keras_learner.py

```diff
@@ -76,13 +76,46 @@ def __init__(self, model: keras.Model,
 
         self.vote_score: float = self.test(self.train_loader)
 
+    def recompile_model(self):
+        """
+        Recompiles the Keras model. This way the optimizer history get erased,
+        which is needed before a new training round, otherwise the outdated history is used.
+        """
+        compile_args = self.model._get_compile_args()  # pylint: disable=protected-access
+        opt_config = self.model.optimizer.get_config()
+
+        if opt_config['name'] == 'Adadelta':
+            compile_args['optimizer'] = keras.optimizers.Adadelta.from_config(opt_config)
+        elif opt_config['name'] == 'Adagrad':
```

Comment:
can we maybe have an automatic version of this? As optimizers can change or extended. someting like `gettatr(keras.optimizes, opt_config['name']).from_config(...`

Reply:
I agree with both comments, fixing them...